File: 251004/Yelkin/4-5/CommentService/src/main/java/by/yelkin/CommentService/dto/CommentRequestTo.java
package by.yelkin.CommentService.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import jakarta.validation.constraints.Size;
import lombok.Data;

@Data
public class CommentRequestTo {
    private Long id;

    @NotNull(message = "topicId should not be null.")
    @Positive(message = "topicId should be a positive number.")
    private Long topicId;

    @NotBlank(message = "Content should not be blank.")
    @Size(min = 2, max = 2048, message = "Content must be between 2 and 2048 characters")
    private String content;
}

File: 251003/Kukhatskavolets/microblog/discussion/src/main/kotlin/by/kukhatskavolets/discussion/dto/requests/CommentRequestTo.kt
package by.kukhatskavolets.discussion.dto.requests

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size

data class CommentRequestTo(
    val id: Long = 0,

    @field:NotNull(message = "Tweet id can't be null")
    val tweetId: Long = 0,

    @field:NotBlank(message = "Content may not be blank")
    @field:Size(min = 2, max = 2048, message = "Content should be between 2 and 2048 symbols")
    val content: String = ""
)